src/v2/ion/IonResponseHelper.js

```diff
@@ -151,7 +152,16 @@ const isIonErrorResponse = (response = {}) => {
   return response.version;
 };
 
+const isTerminalError = (res = {}) => {
+  if (!res.remediations || !Array.isArray(res.remediations) || res.remediations.length !== 1) {
```

Comment:
We should be able to simplify this conditional complexity by removing `!res.remediations`, as `!Array.isArray` will return true/false based on the value existing or not.